test/modules/aws-security-group-integration.ts

```diff
@@ -532,6 +534,103 @@ describe('Security Group Integration Testing', () => {
 
   it('applies the vpc removal', apply());
 
+  // tests cycle in security rules
+  it(
+    'adds a new security group A',
+    query(`  
+    INSERT INTO security_group (description, group_name)
+    VALUES ('Security Group Test A', '${prefix}sgtestA'), ('Security Group Test B', '${prefix}sgtestB');
+  `),
+  );
+
+  it(
+    'adds security group rules pointing to the other ones (A)',
+    query(`
+  INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
+  VALUES ('${prefix}sgtestA', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'),
+  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'), false);
+  `),
+  );
+  it(
+    'adds security group rules pointing to the other ones (B)',
+    query(`
+  INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
+  VALUES ('${prefix}sgtestB', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'),
+  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'), false);
+  `),
+  );
+
+  it('creates the source groups and rules', apply());
+
+  it(
+    'check that security group and rules are correctly created - tcp, udp, icmp',
+    query(
+      `
+      SELECT sg.id, sgr.source_security_group FROM security_group sg INNER JOIN security_group_rule sgr ON sg.id = sgr.security_group_id WHERE sg.group_name='${prefix}sgtestA' AND sgr.source_security_group=(SELECT id FROM security_group WHERE group_name='${prefix}sgtestB')`,
+      (res: any[]) => {
+        expect(res.length).toBe(3);
```

Comment:
Why is the length 3? For this particular query I would only expect one matching row?

Reply:
so when you create a rule with source security, aws expands it to 3 rules... one for tcp, another for udp and another for icmp